src/net_ah_fmt_plug.c

```diff
@@ -0,0 +1,264 @@
+/* Cracker for IPsec Authentication Header (AH) hashes.
+ *
+ * This software is Copyright (c) 2017, Dhiru Kholia <dhiru [at] openwall.com>,
+ * and it is hereby released to the general public under the following terms:
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted.
+ */
+
+#if FMT_EXTERNS_H
+extern struct fmt_main fmt_netah;
+#elif FMT_REGISTERS_H
+john_register_one(&fmt_netah);
+#else
+
+#include <string.h>
+#ifdef _OPENMP
+#include <omp.h>
+#ifndef OMP_SCALE
+#define OMP_SCALE               2048
+#endif
+#endif
+
+#include "formats.h"
+#include "hmacmd5.h"
+#include "misc.h"
+#include "common.h"
+#include "params.h"
+#include "options.h"
+#include "memdbg.h"
+
+#define FORMAT_LABEL            "net-ah"
+#define FORMAT_NAME             "IPsec AH HMAC-MD5-96"
+#define FORMAT_TAG              "$net-ah$"
+#define TAG_LENGTH              (sizeof(FORMAT_TAG) - 1)
+#define ALGORITHM_NAME          "MD5 32/" ARCH_BITS_STR
+#define BENCHMARK_COMMENT       ""
+#define BENCHMARK_LENGTH        0
+#define PLAINTEXT_LENGTH        16  // actual password length limit seems to be 8
```

Comment:
You are right. When I use `1234567£` as the password, it is truncated at after 0xc2.

```
Breakpoint 1, hmac_md5 (... key=key@entry=0x55e127dc2e91 "123456", <incomplete sequence \302>, key_len=key_len@entry=8...) at vrrp_ipsecah.c:35
35	{
(gdb) x/16bx key
0x55e127dc2e91:	0x31	0x32	0x33	0x34	0x35	0x36	0x37	0xc2
0x55e127dc2e99:	0x00	0x00	0x00	0x1a	0x00	0x00	0x00	0xe0
(gdb)
```

This behavior applies for `Keepalived` but I don't know about other implementations. I tried finding other things which use IPsec AH but I didn't find anything suitable for testing.